src/Integration.UnitTests/Suppression/SuppressedIssuesFileStorageTests.cs

```diff
@@ -0,0 +1,372 @@
+
+
+            fileStorage.Update("projectKey", issues);
+
+            directory.Verify(d => d.CreateDirectory(GetTempPath()), Times.Once);
+            file.Verify(f => f.WriteAllText(GetFilePath("projectKey"), JsonConvert.SerializeObject(issues)), Times.Once);
+            logger.Verify(l => l.WriteLine(It.IsAny<string>(), It.IsAny<object[]>()), Times.Never);
+        }
+
+        [TestMethod]
+        public void Get_Should_ReadSingleIssueFromFilePath()
+        {
+            SonarQubeIssue issue = CreateIssue("issueKey");
+            var issues = new List<SonarQubeIssue> { issue };
+
+            var logger = new Mock<ILogger>();
+
+            var directory = new Mock<IDirectory>();
+            var file = new Mock<IFile>();
+            file.Setup(f => f.ReadAllText(GetFilePath("projectKey"))).Returns(JsonConvert.SerializeObject(issues));
+            Mock<IFileSystem> fileSystem = CreateFileSystem(directory, file);
+
+            var fileStorage = new SuppressedIssuesFileStorage(logger.Object, fileSystem.Object);
+
+            var issuesGotten = fileStorage.Get("projectKey").ToList();
+
+            directory.Verify(d => d.CreateDirectory(It.IsAny<string>()), Times.Never);
+            file.Verify(f => f.ReadAllText(GetFilePath("projectKey")), Times.Once);
+            logger.Verify(l => l.WriteLine(It.IsAny<string>(), It.IsAny<object[]>()), Times.Never);
+
+            Assert.AreEqual(1, issuesGotten?.Count);
+            Assert.AreEqual(issue.IssueKey, issuesGotten[0].IssueKey);
+        }
+
+        [TestMethod]
+        public void Get_Should_ReadMultipleIssuesFromFilePath()
+        {
+            SonarQubeIssue issue1 = CreateIssue("issueKey1");
+            SonarQubeIssue issue2 = CreateIssue("issueKey2");
+            var issues = new List<SonarQubeIssue> { issue1, issue2 };
+
+            var logger = new Mock<ILogger>();
+
+            var directory = new Mock<IDirectory>();
+            var file = new Mock<IFile>();
+            file.Setup(f => f.ReadAllText(GetFilePath("projectKey"))).Returns(JsonConvert.SerializeObject(issues));
+            Mock<IFileSystem> fileSystem = CreateFileSystem(directory, file);
+
+            var fileStorage = new SuppressedIssuesFileStorage(logger.Object, fileSystem.Object);
+
+            var issuesGotten = fileStorage.Get("projectKey").ToList();
+
+            directory.Verify(d => d.CreateDirectory(It.IsAny<string>()), Times.Never);
+            file.Verify(f => f.ReadAllText(GetFilePath("projectKey")), Times.Once);
+            logger.Verify(l => l.WriteLine(It.IsAny<string>(), It.IsAny<object[]>()), Times.Never);
+
+            Assert.AreEqual(2, issuesGotten?.Count);
+            Assert.AreEqual(issue1.IssueKey, issuesGotten[0].IssueKey);
+            Assert.AreEqual(issue2.IssueKey, issuesGotten[1].IssueKey);
+        }
+
+        [TestMethod]
+        public void Update_Should_LogWhenProjectKeyIsNull()
```

Comment:
There's a lot of duplication between this test and the "project key is empty" test.
It would be better as a data-driven test (although I think we should be throwing if the project key is invalid).